src/wallet/wallet_tx_builder.cpp

```diff
@@ -14,6 +14,66 @@ int GetAnchorHeight(const CChain& chain, uint32_t anchorConfirmations)
     return nextBlockHeight - anchorConfirmations;
 }
 
+static tl::expected<void, InputSelectionError>
+ValidateAmount(const SpendableInputs& spendable, const CAmount& fee)
+{
+    // TODO: The actual requirement should probably be higher than simply `fee` – do we need to
+    //       take into account the dustThreshold when adding an output? But, this was the
+    //       pre-WalletTxBuilder behavior, so it’s fine to maintain it for now.
+    auto targetAmount = fee;
+    if (spendable.Total() < targetAmount)
+        return tl::make_unexpected(ReportInvalidFunds(spendable, false, fee, 0, targetAmount, 0));
+    else
+        return {};
+}
+
+static tl::expected<std::pair<ResolvedPayment, CAmount>, InputSelectionError>
+ResolveNetPayment(
+        const ZTXOSelector& selector,
+        const SpendableInputs& spendable,
+        const NetAmountRecipient& netpay,
+        const std::optional<CAmount>& fee,
+        const TransactionStrategy& strategy,
+        bool afterNU5)
+{
+    bool canResolveOrchard = afterNU5 && !selector.SelectsSprout();
+    CAmount maxSaplingAvailable = spendable.GetSaplingTotal();
+    CAmount maxOrchardAvailable = spendable.GetOrchardTotal();
+    uint32_t orchardOutputs{0};
+
+    // We initially resolve the payment with `MINIMUM_FEE` so that we can use the payment to
+    // calculate the actual fee.
```

Comment:
This is a reasonable approach because we know that the fee can only increase, so any constraints imposed upon the payment as a consequence of spending some amount can only be diminished if, for example, the fee amount rises to consume some value that was previously intended as output.